### PR TITLE
vendor: cluster-api-provider-baremetal and baremetal-operator

### PR DESCRIPTION
**Pull request by stbenjam**
This updates the URL for the baremetal-operator, uses an openshift
URL instead of metal3-io for cluster-api-baremetal-operator, and pulls
in the latest cluster-api-provider-baremetal.

### CODE REVIEW

**Comment by russellb** [comment]
did you re-vendor it too?

**Comment by stbenjam** [comment]
> did you re-vendor it too?

I have now

File: Gopkg.toml [hunk 1]
```diff
@@ -107,8 +107,9 @@ required = [
 [[constraint]]
   branch = "master"
   name = "github.com/metal3-io/cluster-api-provider-baremetal"
-  source = "github.com/openshift/cluster-api-provider-baremetal/pkg/apis"
+  source = "https://github.com/openshift/cluster-api-provider-baremetal/pkg/apis"
```
Comment (russellb) [inline Gopkg.toml:110]: looks unrelated?

Comment (russellb) [reply Gopkg.toml:110]: looks like CAPBM got updated, but not BMO

Comment (stbenjam) [reply Gopkg.toml:110]: There's no update to BMO, I'm only pulling in the BMC package which hasn't had changes.

Comment (stbenjam) [reply Gopkg.toml:110]: > looks unrelated?

Shouldn't we enforce it to be https?

Comment (russellb) [reply Gopkg.toml:110]: ok, that's fine, the PR title and commit msg just referred to BMO, not this component.

Comment (stbenjam) [reply Gopkg.toml:110]: I've fixed the commit

**Review by russellb — APPROVED** [review]

**Comment by metal3ci** [comment]
Build FAILURE, see build http://10.8.144.11:8080/job/dev-tools/774/

**Comment by stbenjam** [comment]
Kicked off CI again http://10.8.144.11:8080/job/dev-tools/784

**Comment by metal3ci** [comment]
Build SUCCESS, see build http://10.8.144.11:8080/job/dev-tools/784/
